distributed/utils_perf.py

```diff
@@ -181,6 +181,10 @@ def _gc_callback(self, phase, info):
         # don't waste time measuring them
         if info["generation"] != 2:
             return
+        if phase == "stop":
+            # Stop timer before measuring RSS to avoid this influencing the
+            # measurement
+            self._fractional_timer.stop_timing()
         if self._proc is not None:
             rss = self._proc.memory_info().rss
```

Comment:
At the very least if gc runs frequently, this could be expensive. I'm wondering if we want to add yet another timer for this specifically